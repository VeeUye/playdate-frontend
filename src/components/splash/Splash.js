import React from "react";
import { useHistory } from "react-router-dom";
import screenSize from "../../functions/screenSize";
import Subheading from "../atoms/subheading/Subheading";
import Title from "../atoms/title/Title";
import Image from "../../assets/images/swinging.svg";
import Button from "../atoms/Button/index";
import splash from "./splash.module.css";
import button from "../atoms/Button/styles.module.css";
import titleStyles from "../atoms/title/title.module.css";

const Splash = () => {
  const isSmall = screenSize();
  const history = useHistory();

  const handleSignUp = () => {
    history.push("/sign-up");
  };

  const handleSignIn = () => {
    history.push("/sign-in");
  };

  return (
    <>
      <div className={isSmall ? splash.smallScreen : splash.bigScreen}>
        <div className={splash.background}>
          <Title
            className={
              isSmall ? titleStyles.splash__title : titleStyles.splash__bigTitle
            }
            text="Playdate"
          />
          <Subheading />
          <img
            className={isSmall ? splash.img : splash.bigScreenImg}
            src={Image}
            alt="girl on a swing"
          />
          <div
            className={
              isSmall ? splash.buttonWrapper : splash.bigScreenButtonWrapper
            }
          >
            <Button
              className={button.signIn}
              label="Sign In"
              onClick={handleSignIn}
            ></Button>
            <Button
              className={button.signUp}
              label="Sign Up"
              onClick={handleSignUp}
            ></Button>
          </div>
        </div>
      </div>
    </>
  );
};

export default Splash;
