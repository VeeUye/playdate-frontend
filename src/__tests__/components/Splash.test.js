import React from "react";
import { render, screen } from "@testing-library/react";
import Splash from "../../components/splash/Splash";
import Image from "../../assets/images/swinging.svg";

describe("Splash", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Splash />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("displays the title", () => {
    render(<Splash />);

    expect(screen.getByText(/Playdate/).textContent).toBe("Playdate");
  });

  it("displays the tagline", () => {
    render(<Splash />);

    expect(screen.getByText(/Scheduler/i).textContent).toBe(
      "Scheduler for parents"
    );
  });

  it("displays the splash image", () => {
    render(<Splash src={Image} />);

    const image = screen.getByRole("img");

    expect(image).toHaveAttribute("src", "swinging.svg");
    expect(image).toHaveAttribute("alt", "girl on a swing");
    expect(image).toBeVisible();
  });

  it("displays a sign in button", () => {
    render(<Splash />);

    const signInButton = screen.getByRole("button", { name: /sign in/i });

    expect(signInButton).toHaveTextContent(/sign in/i);
  });

  it("displays a sign up button", () => {
    render(<Splash />);

    const signUpButton = screen.getByRole("button", { name: /sign up/i });

    expect(signUpButton).toHaveTextContent(/sign up/i);
  });
});
