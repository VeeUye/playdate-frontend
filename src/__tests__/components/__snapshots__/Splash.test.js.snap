// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Splash renders correctly 1`] = `
<DocumentFragment>
  <div
    class="background"
  >
    <div
      class="splash__title"
    >
      Playdate
    </div>
    <div
      class="text"
    >
      Scheduler for parents
    </div>
    <img
      alt="girl on a swing"
      class="img"
      src="swinging.svg"
    />
    <button
      class="signIn"
    >
      Sign In
    </button>
    <button
      class="signUp"
    >
      Sign Up
    </button>
  </div>
  <div
    class="background"
  >
    <div
      class="background"
    >
      <div
        class="default"
      >
        Create Event
      </div>
      <form>
        <div
          class="field1"
        >
          <div>
            <div
              class="field"
            >
              <label
                class="label"
                for="name"
              >
                Event Name
                <input
                  class="input"
                  id="name"
                  name="name"
                  type="text"
                  value="test"
                />
              </label>
            </div>
            <div
              class="field"
            >
              <label
                class="label"
                for="date"
              >
                Date
                <input
                  class="input"
                  id="date"
                  name="date"
                  type="date"
                  value=""
                />
              </label>
            </div>
            <div
              class="splitInput"
            >
              <div
                class="field"
              >
                <label
                  class="label"
                  for="startTime"
                >
                  Start
                  <input
                    class="input"
                    id="startTime"
                    name="startTime"
                    type="time"
                    value=""
                  />
                </label>
              </div>
              <div
                class="field"
              >
                <label
                  class="label"
                  for="endTime"
                >
                  End
                  <input
                    class="input"
                    id="endTime"
                    name="endTime"
                    type="time"
                    value=""
                  />
                </label>
              </div>
            </div>
            <div
              class="field"
            >
              <label
                class="label"
                for="location"
              >
                Location
                <input
                  class="input"
                  id="location"
                  name="location"
                  type="text"
                  value=""
                />
              </label>
            </div>
            <div
              class="field"
            >
              <label
                class="label"
                for="invite"
              >
                Invite
                <input
                  class="input"
                  id="invite"
                  name="invite"
                  type="email"
                  value=""
                />
              </label>
            </div>
          </div>
        </div>
        <button
          class="createEvent"
        >
          Create Event
        </button>
      </form>
      <img
        alt="girl floating"
        class="img"
        src="float.svg"
      />
    </div>
  </div>
</DocumentFragment>
`;
